time-series/time-series-api/src/main/java/com/powsybl/timeseries/ast/BinaryMaxCalc.java

```diff
@@ -52,6 +52,6 @@ public boolean equals(Object obj) {
 
     @Override
     public int hashCode() {
-        return left.hashCode() + right.hashCode();
+        return hash;
```

Comment:
There is a problem here, and in the other classes extending `AbstractBinaryNodeCalc`.

`hash` is only computed in the constructors of these classes, but `left` and `right` are not final, so they can change.
If `setLeft(...)` or `setRight(...)` are called, the hash doesn't match anymore with the content.

This leads to a major problem: `hashCode()` doesn't respect its general contract:

> If two objects are equal according to the [equals](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Object.html#equals(java.lang.Object)) method, then calling the hashCode method on each of the two objects must produce the same integer result.

([source](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Object.html#hashCode()))

For instance, the following test fails on the last assertion:

```java
    @Test
    void hashTest() {
        NodeCalc node1 = new IntegerNodeCalc(1);
        NodeCalc node2 = new IntegerNodeCalc(2);
        NodeCalc node3 = new IntegerNodeCalc(3);

        AbstractBinaryNodeCalc binary1 = new BinaryMaxCalc(node1, node3);
        binary1.setRight(node2);
        NodeCalc binary2 = new BinaryMaxCalc(node1, node2);

        assertEquals(binary2, binary1);
        assertEquals(binary2.hashCode(), binary1.hashCode());
    }
```